hash-backend/src/interpreter/command.rs

```diff
@@ -0,0 +1,69 @@
+//! Hash interactive mode commands
+//
+// All rights reserved 2021 (c) The Hash Language authors
+
+use crate::interpreter::error::{report_interp_error, InterpreterError};
+
+#[derive(Debug, Clone)]
+pub enum InteractiveCommand {
+    /// Quit the current interactive session
+    Quit,
+    /// Clear the console
+    Clear,
+    /// Get the type of the expression
+    // @@Incomplete: This variant should store an AstNode<Expression> so that we can use the typer to get the type from the passed expression
+    Type(String),
+    /// Just prints the version of the current interactive mode
+    Version,
+}
+
+impl InteractiveCommand {
+    /// Attempt to convert a string into an interactive command
+    pub fn from(input: &str) -> Option<InteractiveCommand> {
+        match input {
+            ":q" => Some(InteractiveCommand::Quit),
+            ":c" | ":cls" | ":clear" => Some(InteractiveCommand::Clear),
+            ":v" => Some(InteractiveCommand::Version),
+            _ => {
+                if input.len() > 3 {
+                    let (command, arg) = input.split_at(3);
+                    // check here if it begins with either a ':t' or a ':p' and then take an expression
+                    if command == ":t " {
+                        return Some(InteractiveCommand::Type(String::from(arg)));
+                    }
+                }
+
+                None
```

Comment:
Can be simplified to
```rust
_ => input.strip_prefix(":t ").map(|arg| InteractiveCommand::Type(arg.to_owned()))
```

Also, is `:p` in the comment `check here if it begins with either a ':t' or a ':p' and then take an expression` inaccurate?

Reply:
I think it is better to do all the command checking in here and return some sort of `Result` type?

Reply:
Oh and then return an `InterpreterError` if it fails, then we could get rid of the `report_error` function?